zeppelin-zengine/src/main/java/org/apache/zeppelin/scheduler/RemoteScheduler.java

```diff
@@ -53,22 +55,26 @@ public void runJobInScheduler(Job job) {
             remoteInterpreter.getProperty(".execution.mode", "paragraph");
     if (executionMode.equals("paragraph")) {
       // wait until it is submitted to the remote
-      while (!jobRunner.isJobSubmittedInRemote()) {
+      while (!jobRunner.isJobSubmittedInRemote() && !Thread.currentThread().isInterrupted()) {
```

Comment:
The RemoteScheduler waits until the note or paragraph is executed on the remote interpreter.
If an interrupt should occur, it is now reacted to and aborted accordingly.
The executor of the RemoteScheduler receives the request to terminate itself through the `stop` in the AbstractScheduler.